delta/sourcing/src/main/scala/ch/epfl/bluebrain/nexus/sourcing/processor/ProcessorCommand.scala

```diff
@@ -1,139 +1,92 @@
-    *
-    * @param event the appended
-    * @param state the state generated from the appended event
-    */
-  final case class AppendSuccess[Event, State] private[processor] (id: String, event: Event, state: State)
-      extends EventSourceCommand
-
-  /**
-    * Defines a command to evaluate the command giving a [[EvaluationResult]]
-    * which is sent back to the replyTo
-    *
-    * @param command the command to evaluate
-    * @param replyTo the actor to send the result to
-    */
-  final case class Evaluate[Command](id: String, command: Command, replyTo: ActorRef[EvaluationResult])
-      extends InputCommand
-
-  /**
-    * Same thing as [[Evaluate]] but the result is never appended to the event log
-    * @param command the command to test
-    * @param replyTo the actor to send the result to
-    */
-  final case class DryRun[Command](id: String, command: Command, replyTo: ActorRef[DryRunResult]) extends InputCommand
-
-  // Evaluation results
-  sealed trait RunResult        extends ProcessorCommand
-  sealed trait EvaluationResult extends RunResult
+    sealed trait ReadOnlyRequest extends AggregateRequest
+    final case class RequestState[State](id: String, replyTo: ActorRef[AggregateResponse.StateResponse[State]]) extends ReadOnlyRequest
+    final case class RequestLastSeqNr(id: String, replyTo: ActorRef[AggregateResponse.LastSeqNr]) extends ReadOnlyRequest
+  }
 
   /**
-    * Describes the event and the state computed when an an [[Evaluate]] or a [[DryRun]]
-    * has been successfully evaluated
-    *
-    * @param event the generated event
-    * @param state the new state
+    * Message issued when when passivation is triggered
     */
-  final case class EvaluationSuccess[Event, State](event: Event, state: State) extends EvaluationResult
+  final private[processor] case object Idle extends ProcessorCommand
 
   /**
-    * Describes the event resulting from a successful evaluation.
-    *
-    * @param event the generated event
+    * Outgoing messages from child actor to the [[EventSourceProcessor]] actor
     */
-  final private[processor] case class EvaluationSuccessInternal[Event](event: Event) extends EvaluationResult
+  sealed private[processor] trait ChildActorResponse extends ProcessorCommand
+  private[processor] object ChildActorResponse {
+    final case class AppendResult[Event, State](event: Event, state: State) extends ChildActorResponse
+    final case class StateResponseInternal[State](value: State)             extends ChildActorResponse
+  }
 
   /**
-    * Rejection that occured after running an an [[Evaluate]] or a [[DryRun]]
-    * @param value the rejection
+    * Messages issued from within the [[EventSourceProcessor]] as an evaluation result
     */
-  final case class EvaluationRejection[Rejection](value: Rejection) extends EvaluationResult
+  sealed private[processor] trait EvaluationResult extends ProcessorCommand
```

Comment:
Maybe we can call them EvaluationResultInternal to make them more distinguishable from the actual response ?